From cd1722a8deeacc831898cdf380528b9bc5405a12 Mon Sep 17 00:00:00 2001
From: Vinaykumar Masane <vinaykumar.masane@freescale.com>
Date: Mon, 27 Jul 2015 17:59:04 +0530
Subject: [PATCH 11/60] SBR-19

Addressed comments from Customer to maintain string in UPPER Case and
internal review inputs.

Signed-off-by: Vinaykumar Masane <vinaykumar.masane@freescale.com>
---
 commands/boardinfo.c | 86 +++++++++++++++++++++++++++++++++++-----------------
 1 file changed, 58 insertions(+), 28 deletions(-)

diff --git a/commands/boardinfo.c b/commands/boardinfo.c
index 2143086..508ac1e 100644
--- a/commands/boardinfo.c
+++ b/commands/boardinfo.c
@@ -36,6 +36,18 @@
 #include <errno.h>
 #include <environment.h>
 
+#define sizePID 11  /* Max 11 Character Product ID */
+#define sizeVID  3  /* Max  3 Character H/W Version */
+#define sizeS_N 11  /* Max 11 Character Board Serial Number */
+
+#define countMAC        3 /* 3 MAC-IDs one each for LAN, WAN-1, WAN-2 */
+#define totalMAC_OCTET  6 /* No. of MAC Octets in MAC ID */
+
+#define keyCR 0x0D  /* Enter Key Press */
+
+#define offsetALPHA 0x37
+#define offsetDIGIT 0x30
+
 typedef struct c2krv340_board_info {
   char pid[12];             /* Product ID */
   char vid[4];              /* H/W Version */
@@ -50,8 +62,14 @@ char *filename = "/dev/spi1.board-info" ;
 static int command_boardinfo_update(int argc, char *argv[])
 {
   int ret = 0, fd = -1, n_written = 0;
-  int i, len = 0, isOctetFull = 0, n_mac_octet = 0;
+  int i, len = 0, n_mac_octet = 0;
   char c = '\0';
+  enum mac_octet_state {
+    empty = 0,
+    half  = 1,
+    full  = 2
+  };
+  enum mac_octet_state isOctetFull = empty;
   c2krv340 board_data;
 
   flush(stdin);
@@ -59,17 +77,20 @@ static int command_boardinfo_update(int argc, char *argv[])
   memset(&board_data, 0x00, sizeof(board_data));
 
   printf("\nEnter PID : ");
-  while (len < sizeof(board_data.pid))
+  while (len < sizePID)
   {
     c = getc();
-    if ( isalnum(c) || c == 0x0A || c == 0x0D || c == 0x2D )
+    if ( isalnum(c) || c == keyCR || c == '-' )
+    {
+      c = toupper(c);
       putchar(c);
+    }
     else
     {
       printf("\n c = %c : 0x%X Input Error! %s", c, c, help_string);
       return COMMAND_ERROR;
     }
-    if ( c == 0x0D )
+    if ( c == keyCR )
     {
       break;
     }
@@ -78,17 +99,20 @@ static int command_boardinfo_update(int argc, char *argv[])
 
   len = 0;
   printf("\nEnter VID : ");
-  while (len < sizeof(board_data.vid))
+  while (len < sizeVID)
   {
     c = getc();
-    if ( isalnum(c) || c == 0x0A || c == 0x0D )
+    if ( isalnum(c) || c == keyCR )
+    {
+      c = toupper(c);
       putchar(c);
+    }
     else
     {
       printf("\n c = %c : 0x%X Input Error! %s", c, c, help_string);
       return COMMAND_ERROR;
     }
-    if ( c == 0x0D )
+    if ( c == keyCR )
     {
       break;
     }
@@ -97,51 +121,54 @@ static int command_boardinfo_update(int argc, char *argv[])
 
   len = 0;
   printf("\nEnter S/N : ");
-  while (len < sizeof(board_data.s_n))
+  while (len < sizeS_N)
   {
     c = getc();
-    if ( isalnum(c) || c == 0x0A || c == 0x0D || c == 0x2D )
+    if ( isalnum(c) || c == keyCR )
+    {
+      c = toupper(c);
       putchar(c);
+    }
     else
     {
       printf("\n c = %c : 0x%X Input Error! %s", c, c, help_string);
       return COMMAND_ERROR;
     }
-    if ( c == 0x0D )
+    if ( c == keyCR )
     {
       break;
     }
     board_data.s_n[len++] = c;
   }
 
-  for (i=0; i<3; i++)
+  for (i = 0; i < countMAC; i++)
   {
-    isOctetFull = 0;
+    isOctetFull = empty;
     n_mac_octet = 0;
     printf("\nEnter %-5s MAC : ",  i == 0 ? "LAN" : (i == 1 ? "WAN-1" : "WAN-2"));
-    while (n_mac_octet < 6)
+    while (n_mac_octet < totalMAC_OCTET)
     {
-      c = getc();
+      c = toupper(getc());
       putchar(c);
-      if ((isxdigit(c)) && ( isOctetFull < 2 ))
+      if ((isxdigit(c)) && ( isOctetFull < full ))
       {
         unsigned char offset = 0;
-        offset = islower(c) ? 0x57 : isupper(c) ? 0x37 : isdigit(c) ? 0x30 : 0;
-        if ( isOctetFull == 0 )
+        offset = isupper(c) ? offsetALPHA : isdigit(c) ? offsetDIGIT : 0;
+        if ( isOctetFull == empty )
         {
           board_data.mac[i][n_mac_octet] = ((unsigned int)c - offset) << 4;
-          isOctetFull = 1;
+          isOctetFull = half;
         }
-        else if ( isOctetFull == 1 )
+        else if ( isOctetFull == half )
         {
           board_data.mac[i][n_mac_octet] |= (((unsigned int)c - offset) & 0xFF);
           ++n_mac_octet;
-          isOctetFull = 2;
+          isOctetFull = full;
         }
       }
-      else if ((  c == ':' ) && ( isOctetFull == 2 ))
+      else if ((  c == ':' ) && ( isOctetFull == full ))
       {
-        isOctetFull = 0;
+        isOctetFull = empty;
         continue;
       }
       else
@@ -211,23 +238,25 @@ static int command_boardinfo_show(int argc, char *argv[])
   flush(stdout);
 
   printf("\nPID  : ");
-  for(i = 0; i < 12; i++)
+  for(i = 0; i < sizePID; i++)
   {
     if (isprint(*((char *)board_data->pid + i)))
       putchar(*((char *)board_data->pid + i));
     else
       putchar(' ');
   }
+
   printf("\nVID  : ");
-  for(i = 0; i < 4; i++)
+  for(i = 0; i < sizeVID; i++)
   {
     if (isprint(*((char *)board_data->vid + i)))
       putchar(*((char *)board_data->vid + i));
     else
       putchar(' ');
   }
+
   printf("\nS/N  : ");
-  for(i = 0; i < 12; i++)
+  for(i = 0; i < sizeS_N; i++)
   {
     if (isprint(*((char *)board_data->s_n + i)))
       putchar(*((char *)board_data->s_n + i));
@@ -236,7 +265,7 @@ static int command_boardinfo_show(int argc, char *argv[])
   }
   flush(stdout);
 
-  for (i=0; i<3; i++)
+  for (i = 0; i < countMAC; i++)
   {
     memset(out_str, 0x00, sizeof(out_str));
     sprintf(out_str,"%02X:%02X:%02X:%02X:%02X:%02X",\
@@ -248,6 +277,7 @@ static int command_boardinfo_show(int argc, char *argv[])
                       (unsigned char)board_data->mac[i][5]);
     printf("\n%-5s MAC : %s", i == 0 ? "LAN" : i == 1 ? "WAN-1" : "WAN-2", out_str);
   }
+
   free(board_data);
   printf("\n");
   return COMMAND_SUCCESS;
@@ -277,8 +307,8 @@ static const __maybe_unused char cmd_boardinfo_help[] =
 " -u      update board-info (default updates info to /dev/spi1.board-info)\n"
 " -s      lists boardinfo contents from /dev/spi1.board-info\n"
 " e.g.    boardinfo -u\n"
-"Enter PID : <MAX 12 char>\n"
-"Enter VID : <MAX 4 char>\n"
+"Enter PID : <MAX 11 char>\n"
+"Enter VID : <MAX  3 char>\n"
 "Enter S/N : <MAX 11 char>\n"
 "Enter LAN MAC   : <6 octects of MAC address with : separator e.g. aa:bb:cc:dd:ee:f0 >\n"
 "Enter WAN-1 MAC : <6 octects of MAC address with : separator e.g. aa:bb:cc:dd:ee:f1 >\n"
-- 
2.7.4

